### PR TITLE
feat: Core logic for SARIF adapter

### PR DESCRIPTION
**Pull request by petertrr**

### CODE REVIEW

File: src/main/kotlin/org/cqfn/save/adapter/Adapter.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import io.github.detekt.sarif4k.Version
+import kotlinx.serialization.StringFormat
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.serializer
+import java.io.InputStreamReader
+import java.io.OutputStreamWriter
+import kotlin.reflect.KType
+
+abstract class Adapter<T : Any>(
```
Comment (orchestr7) [inline src/main/kotlin/org/cqfn/save/adapter/Adapter.kt]: extract core logic please

Comment (orchestr7) [inline src/main/kotlin/org/cqfn/save/adapter/Adapter.kt]: rename T to IN

Comment (orchestr7) [inline src/main/kotlin/org/cqfn/save/adapter/Adapter.kt]: `T` should implement some Interface?

Comment (petertrr) [reply src/main/kotlin/org/cqfn/save/adapter/Adapter.kt]: Having factory method makes this not required: if we need to set type only once, there is no need to fool-proof it with marker interface. We also didn't plan any useful features for this interface, except for marker.

Comment (petertrr) [reply src/main/kotlin/org/cqfn/save/adapter/Adapter.kt]: Done

Comment (petertrr) [reply src/main/kotlin/org/cqfn/save/adapter/Adapter.kt]: Split into `Adapter`, `KxSerializationAdapter`, `SarifAdapter` and created a factory method

**Pull request merged**